docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,247 @@
+uWSGI can find it; we will do that by creating a ``wsgi.py`` file. Modify the
+content of this file based on how your application is configured.  Ensure that
+you expose  a variable called ``application`` which is assigned to your
+:any:`falcon.API` instance.
+
+.. code-block:: python
+  :caption: /home/myproject/src/wsgi.py
+
+  import os
+  import myproject
+
+  config = myproject.get_config(os.environ['MYPROJECT_CONFIG'])
+  application = myproject.create_api(config)
+
+The above example shows how to implement a bare-bones ``wsgi.py``. It imports your
+application and creates the :any:`falcon.API` instance that represents our
+application, exposing it as the module-scoped ``application`` variable. uWSGI
+will use this file to load your application.
+
+Note that we did not call ``run`` like a developer might do during development
+or use `wsgiref.simple_server.make_server`.  Instead we simply assigned our
+`API` instance to ``application``. We *do not* want to start a server, since
+uWSGI does that automatically. Starting an independent WSGI server in your
+``wsgi.py`` file will render unexpected results.
+
+
+Deploying Falcon behind uWSGI
+'''''''''''''''''''''''''''''
+
+With our ``wsgi.py`` file in place, it is time to configure uWSGI.  To do this,
+we create a ``uwsgi.ini`` file. In general, you shouldn’t commit this file to
+source control; it should be generated from a template by your deployment
+toolchain according to the target environment (number of CPUs, etc.).
+
+This configuration, when executed, will create a new uWSGI server pointed at our
+wsgi.py file and listening at ``12.0.0.1:8080``.
+
+.. code-block:: ini
+  :caption: /home/myproject/src/uwsgi.ini
+
+  [uwsgi]
+  master = 1
+  vacuum = true
+  socket = 127.0.0.1:8080
+  enable-threads = true
+  thunder-lock = true
+  threads = 2
+  processes = 2
+  virtualenv = /home/myproject/venv
+  wsgi-file = /home/myproject/src/wsgi.py
+  chdir = /home/myproject/src
+  uid = myproject-runner
+  gid = myproject-runner
+
+
+.. note:: Thread vs Processes
+
+  There are many questions to consider when deciding how to manage the processes
+  that actually run your Python code. Are you generally CPU bound or IO bound?
+  Is your application thread-safe? How many CPU's do you have? What system are
+  you on? Do you need an in-process cache?
+
+  The configuration presented here enables both threads and processes. However,
+  you will have to experiment and do some research to understand your
+  application's unique requirements, and then tailor your uWSGI configuration
+  accordingly. Generally speaking, uWSGI is flexible enough to support most
+  types of applications.
+
+.. note:: TCP or Unix Sockets
+
+  NGINX and uWSGI can communicate via normal TCP (using an IP address) or Unix
+  sockets (using a socket file). TCP sockets are easier to setup and generally
+  work for simple deployments. If you want to have finer control over what
```

Comment:
We could add this later, but another important advantage of Unix sockets is better performance (lower latency, higher throughput).